helix-core/src/main/java/org/apache/helix/monitoring/mbeans/InstanceMonitor.java

```diff
@@ -117,28 +164,62 @@ public synchronized void updateInstance(Set<String> tags,
       _tags = Lists.newArrayList(tags);
       Collections.sort(_tags);
     }
-    _disabledPartitions = 0L;
+    long numDisabledPartitions = 0L;
     if (disabledPartitions != null) {
       for (List<String> partitions : disabledPartitions.values()) {
         if (partitions != null) {
-          _disabledPartitions += partitions.size();
+          numDisabledPartitions += partitions.size();
         }
       }
     }
     // TODO : Get rid of this when old API removed.
     if (oldDisabledPartitions != null) {
-      _disabledPartitions += oldDisabledPartitions.size();
+      numDisabledPartitions += oldDisabledPartitions.size();
     }
-    _isUp = isLive;
-    _isEnabled = isEnabled;
+
+    _onlineStatusGauge.updateValue(isLive ? 1L : 0L);
+    _enabledStatusGauge.updateValue(isEnabled ? 1L : 0L);
+    _disabledPartitionsGauge.updateValue(numDisabledPartitions);
   }
 
   /**
    * Increase message received for this instance
    * @param messageReceived received message numbers
    */
   public synchronized void increaseMessageCount(long messageReceived) {
-    _totalMessageReceived += messageReceived;
+    _totalMessagedReceivedCounter
+        .updateValue(_totalMessagedReceivedCounter.getValue() + messageReceived);
   }
 
+  /**
+   * Update max capacity usage for this instance.
+   * @param maxCapacityUsage max capacity usage of this instance
+   */
+  public synchronized void updateMaxCapacityUsage(double maxCapacityUsage) {
+    _maxCapacityUsageGauge.updateValue(maxCapacityUsage);
+  }
+
+  /**
+   * Get max capacity usage of this instance.
+   * @return Max capacity usage of this instance.
+   */
+  public synchronized double getMaxCapacityUsageGauge() {
+    return _maxCapacityUsageGauge.getValue();
+  }
```

Comment:
Not required unless for test.

Reply:
Wanted to make it consistent with other gauges' APIs. We can put it protected if we want to keep this api.

Reply:
This is used in instance monitor test. In that test, the monitor is not registered to mbean server. So it is for api/value  update test.